File: list.cpp
#include<iostream>
using namespace std;

class Node {
    public:
    int data;
    Node *next;
};
class List{
    public:
    Node *head;
    int _length;
    init(int data){
        head = new Node();
        head->data = data;
        head->next = NULL;
        _length = 1;
    }
    void add(int data){
        Node *ptr,*new_node;
        ptr = head;
        while(ptr->next != NULL){
            ptr = ptr->next;
        }
        new_node = new Node();
        new_node->data = data;
        new_node->next = NULL;
        ptr->next=new_node;
        _length += 1;
    }
    void print(){
        Node *ptr;
        ptr = head;
        while(ptr != NULL){
            cout << "[" << ptr->data << "]->";
            ptr = ptr->next;
        }
        cout << " ";
    }
    void insertstart(int data){
        Node *ptr = head;
        Node *new_node = new Node();
        new_node->data = data;
        new_node->next = ptr;
        head = new_node;
        _length += 1;
    }
    void sortAsc(){
        Node *ptr,*iMin,*i,*j;
        i = head;
        while(i != NULL){
            j = i->next;
            iMin = i;
            while(j != NULL){
                if(j->data<iMin->data){
                    iMin =j;
                }
                j = j->next;
            }
            int temp = iMin->data;
            iMin->data = i->data;
            i->data = temp;
            i = i->next;
        }
    }
    void sortDesc(){
        Node *ptr,*iMax,*i,*j;
        i = head;
        while(i != NULL){
            j = i->next;
            iMax = i;
            while(j != NULL){
                if(j->data>iMax->data){
                    iMax =j;
                }
                j = j->next;
            }
            int temp = iMax->data;
            iMax->data = i->data;
            i->data = temp;
            i = i->next;
        }
    }
    bool isEmpy(){
        return head != NULL ? true:false;
    }
    void deleteFirst(){
        if(head->next != NULL){
            Node *ptr = head->next;
            delete head;
            head = ptr;
            _length -= 1;
        }
    }
    void reverse(){
        Node *j,*ptr; 
        ptr = head;
        while(ptr != NULL){
            j = ptr->next;
            while(j != NULL){
                int temp = j->data;
                j->data = ptr->data;
                ptr->data = temp;
                j = j->next;
            }
            ptr = ptr->next;
        }

    }
    int length(){
        return _length;
    }
};

int main(){
    List l = List();
    l.init(0);
    l.add(2);
    l.add(3);
    l.add(4);
    l.add(5);
    l.add(6);
    l.add(7);
    l.print();
    cout<<endl;
    l.reverse();
    l.print();
    cout<<endl;
    l.sortDesc();
    l.print();
    cout << endl;
    cout << l.isEmpy();
    return 0;
}

File: list.js
class Node{
    constructor(data){
        this.data = data 
        this.next =  null
    }
}
class List{
    constructor(data){
        this.head = new Node(data)
        this._length = 1
    }
    print(){
        let ptr = this.head 
        while(ptr != null){
            console.log(`[${ptr.data}]`)
            ptr = ptr.next
        }
    }
    add(data){
        let ptr = this.head 
        while(ptr.next != null){
            ptr = ptr.next
        }
        ptr.next = new Node(data)
        this._length += 1
    }

    deleteFirst(){
        this.head = this.head.next
        this._length -= 1
    }
    isEmpty(){
        return this.head != null ? false : true
    }
    reverse(){
       let ptr = this.head 
       while(ptr != null){
          
           let j =ptr.next 
           while(j != null){
               let temp = j.data 
               j.data = ptr.data 
               ptr.data = temp 
               j = j.next
           }
           ptr = ptr.next
       }
    }
    sortAsc(){
        let i = this.head
        while(i != null){
            let j = i.next 
            let iMin = i 
            while(j!=null){
                if(iMin.data>j.data){
                    iMin = j
                }
                j = j.next
            }
            let temp = iMin.data 
            iMin.data = i.data 
            i.data = temp 
            i = i.next
        }
    }
    sortDesc(){
        let i = this.head
        while(i != null){
            let j = i.next 
            let iMax = i 
            while(j!=null){
                if(iMax.data<j.data){
                    iMax = j
                }
                j = j.next
            }
            let temp = iMax.data 
            iMax.data = i.data 
            i.data = temp 
            i = i.next
        }
    }
    find(value){
        let i = this.head
        while(i!=null){
            if(i.data==value){
                return true
            }
            i = i.next
        }
        return false
    }
    length(){
        return this._length
    }
}


window.setTimeout(()=>{
    a = new List(10)

for(let i=0;i<10;i++){
    let temp = parseInt(prompt("entrer lage "+i))
    a.add(temp)
}
let state = true
while(state){
    let choix = parseInt(prompt("entrer votre choix :"));
    console.log(typeof choix)
    switch(choix){
        case 1:
            console.clear()
            a.print()
            break 
        case 2:
            console.clear()
            a.sortAsc()
            a.print()
            break
        case 3:
            console.clear()
            a.sortDesc()
            a.print()

            break
        case 4:
            conosle.clear()
            a.reverse()
            a.print()
            break
        case 5:
            console.clear()
            a.deleteFirst()
            a.print()
            break
        default:
            state = false
            break

    }
}

},3000)
